assets/src/application/services/notification/index.ts

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Icon, notification as defaultNotification } from 'antd';
+import { ArgsProps } from 'antd/lib/notification';
+
+import error from './error';
+
+const getHash = (message): string => {
+	const msg = JSON.stringify(message);
+	return msg
+		.split('')
+		.reduce((a, b) => {
+			a = (a << 5) - a + b.charCodeAt(0);
+			return a & a;
+		}, 0)
+		.toString();
+};
```

Comment:
this is likely going to be problematic because the AntD typing has the `message` prop set as a `ReactNode` whereas this is expecting a `string` (and will likely blow up if a `ReactNode` is run through `.split('')`). So we might have to use something else like a timestamp for the message keys.

Reply:
Used `uuid` instead: https://github.com/eventespresso/event-espresso-core/pull/2403/commits/1829b9e4f691e65fee7b6d2a7c3ad5a7822a8e1e